packages/gatsby/src/commands/build-utils.ts

```diff
@@ -70,6 +70,11 @@ function normalizePagePath(path: string): string {
   if (path === `/`) {
     return `/`
   }
+
+  if (platform() === `win32` || platform() === `darwin`) {
```

Comment:
Maybe move the check outside of this function? Platforms likely won't change 😅

Reply:
Done in https://github.com/gatsbyjs/gatsby/pull/31071/commits/495e64330b583dde72f3a390eb75721ebae4e27a